File: pCommitter/input/committermain.py
import logging
import username_provider
import validator
from committer import *

logging.basicConfig(filename='committer.log',
				filemode='a',
				format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
				datefmt='%H:%M:%S',
				level=logging.DEBUG)

def Main():
	username = username_provider.getUsername()

	print(f"Hello {username}, please enter your commit message: (leave empty to view history)")
	print(">>>", end=' ')

	msg = input()

	committer = Committer(None)
	if len(msg) > 0:
		if validator.isValidCommitMsg(msg):
			committer.commitWithMsg(msg)
			print("Changes commited with message")
			logging.info(f"{msg} entered by {username}")
		else:
			print("Please follow the commit message rules... Aborting commit.")
			logging.warning(f"{msg} attempted by {username}")
			return

	committer.showGreetings()

Main()

File: ejCommitter/logging_central/src/main/java/CommitterMain.java
import java.util.logging.Level;
//import java.util.logging.Logger;
import java.util.logging.FileHandler;
import java.util.logging.SimpleFormatter;
import java.io.IOException;
import java.lang.RuntimeException;
import java.util.Scanner;
import org.apache.logging.log4j.ThreadContext;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

final class CommitterMain
{
    private static final Logger logger = LogManager.getLogger("main");

    public static void main(String[] a_args) throws IOException
    {
    	{ // scope for username
	        String username = UsernameProvider.getUsername();
	        ThreadContext.put("user", username);
	    	
	        System.out.print(String.format("Hello %s, please enter your commit message: (leave empty to view history)\n", username));
	        System.out.print(">>> ");
    	}
    
        final String msg;
        try (final Scanner in = new Scanner(System.in))
        {
            msg = in.nextLine();
        }

        Committer committer = new Committer(logger);
        if (!msg.isEmpty()) {
	        if (Validator.IsValidCommitMsg(msg))
	        {
                committer.CommitWithMsg(msg);
                System.out.print("Changes commited with message\n");
                logger.info(String.format("%s", msg));
	        }
	        else
	        {
	            System.out.print("Please follow the commit message rules... Aborting commit.\n");
	            logger.warn(String.format("%s", msg));
	            return;
	        }
        }

        committer.ShowPreviousCommits();    
    }
}
